explorer-ui/src/components/ProfileView/Preferences/Preferences.jsx

```diff
@@ -14,176 +14,257 @@ import Geocode from "react-geocode";
+      visibility: visibility,
+    });
+    console.log("visibility is: ", visibililityChoice);
+  }
+  // Update the state array for the users attractions interests
+  function handleLocationSet(event, region) {
     event.preventDefault();
-    const preferences = async () => {
-      try {
-        const res = await axios.post(
-          `${config.API_BASE_URL}/profileInfo`,
-          {
-            preferenceInfo: preferenceInfo
-          },
-          { maxContentLength: Infinity, maxBodyLength: Infinity }
-        );
-      } catch (err) {
-        alert(err);
-        console.log(err);
+    setCurrLocation(region.current.value);
+    setPreferenceInfo({
+      interests: preferenceInfo.interests,
+      attractions: preferenceInfo.attractions,
+      occupation: preferenceInfo.employment,
+      budget: preferenceInfo.budget,
+      currLocation: currLocation,
+      visibility: preferenceInfo.visibility,
+    });
+    // Get latitude & longitude from address.
+    Geocode.fromAddress(region.current.value).then(
+      (response) => {
+        const { lat, lng } = response.results[0].geometry.location;
+        setPosition({ lat: lat, lng: lng });
+        console.log(lat, lng);
+      },
+      (error) => {
+        console.error(error);
       }
-    };
-    preferences();
-    console.log("preferences posted!")
-  };
+    );
+
+    console.log("current location is: ", currLocation);
+  }
 
+//   // Send the profile information to the database
+//   function handleSubmit(event) {
+//     console.log("submitted");
+//     event.preventDefault();
+//     const preferences = async () => {
+//       try {
+//         const res = await axios.post(
+//           `${config.API_BASE_URL}/profileInfo`,
+//           {
+//             preferenceInfo: preferenceInfo,
+//           },
+//           { maxContentLength: Infinity, maxBodyLength: Infinity }
+//         );
+//       } catch (err) {
+//         alert(err);
+//         console.log(err);
+//       }
+//     };
+//     preferences();
+//     console.log("preferences posted!");
+//   }
 
   // RETURN
   return (
     <div className="preferences">
-        <NavBar />
-      <form className="questionaire" onSubmit={(event) =>handleSubmit(event)}>
+      <NavBar />
+      <div className="questionaire">  
+      {/* onSubmit={(event) => handleSubmit(event)}> */}
         <div className="employment">
           <h3>What best describes the field you work in?</h3>
-          <select id="occupation" name="occupation" ref={employmentOption} onClick={(event) => handleOccupationsSelected(event, employmentOption)}>
-          {
-            employment_fields.map((occupation, idx) =>(
-                <option >{occupation}</option>
-            ))
-          }
+          <select
+            id="occupation"
+            name="occupation"
+            ref={employmentOption}
+            onClick={(event) =>
+              handleOccupationsSelected(event, employmentOption)
```

Comment:
for onClick handlers -> react uses 'synthetic' events, so most of the time will not need to call preventDefault on the event !